src/__tests__/typescript-typedef.spec.ts

```diff
@@ -0,0 +1,49 @@
+import { SchemaComposer } from '../index';
+import { TypeComposer } from '../TypeComposer';
+
+interface Person {
+    uid: string;
+    name: string;
+    nickName: string;
+    age: number;
+}
+
+interface Context {
+    req: any;
+    res: any;
+    uid: string;
+}
+
+// works for cases where u create your instance of
+const schemaComposer = new SchemaComposer<Context>();
+
+// Person from this schemaComposer gets the context defined for that schemaComposer
+const Person = schemaComposer.getOrCreateTC<Person>('Person');
+
+// create from default schemaComposer uses the context type passed in
+// or uses `any` as type
+const PersonTC = TypeComposer.create<Person, Context>({
+    name: 'Person',
+    fields: {
+        name: 'String',
+        nickName: 'String',
+        age: 'Int'
+    }
+});
+
+PersonTC.addResolver({
```

Comment:
Inside `addResolver()` method the `source` cannot be equal to `Person`.
How I said before, `resolvers` are fieldConfigs for the external types.

Need to change this test case for `setFields({ ... })`.
`resolvers` should be unaffected by provided `TSource` to TypeComposer (it still should be any for better times ;)).